funtik-scaffolding/src/main/scala/ru/primetalk/funtik/environment/Controller.scala

```diff
@@ -11,13 +14,23 @@ class Controller
   val ctx: dom.CanvasRenderingContext2D,
   val gameView: View[Display[Boolean]]
 ) {
-  private val rnd = new Random()
+  private val rnd = new ScalaRandom()
   private var currentGameState: Display[Boolean] = generateDisplay(rnd.nextInt())
 
   def generateDisplay(n: Int): Display[Boolean] = {
-    Display.showPoints(Seq((1,1), (79,59)), true, false)
+    val rect = Rectangle(0 -> 0, 79 -> 59)
+    val randoms = Random.stream(System.currentTimeMillis())
+    val roomTree = new BSPTree(10).generate(rect).runA(randoms).value
+    val points = toPoints(roomTree)
+    Display.showPoints(points, true, false)
   }
 
+  private def toPoints(tree: Tree[Rectangle]): List[Position] = tree match {
+    case Node(l, r) => toPoints(l) ++ toPoints(r)
```

Comment:
`++` - is expensive.
(1) what about using accum?
(2) we may use `reverse_:::` because we don't really care about order.

Reply:
Ok